app/src/ingest_policy_pdfs.py

```diff
@@ -106,11 +104,13 @@ def process_chunk(text: str, document: Document, db_session: db.Session) -> None
             current_chunk.append(sentence)
```

Comment:
I think the logic in lines 101-103 might be incorrect. Should it be this?:
```
        total_token_count = len(embedding_model.tokenizer.tokenize("".join(current_chunk)+sentence))

        if total_token_count <= embedding_model.max_seq_length:
```

Also to confirm: we're comparing "tokens" to "seq_length" -- are they comparable?

Reply:
I'm thinking they are not comparable.
* a token is a string, typically a word or part of a word
* `max_seq_length` refers to the length of a vector of floats, where each float could reflect a word, several words, or a concept.

Reply:
Sorry, didn't see your comment as I was writing mine

> max_seq_length refers to the length of a vector of floats, where each float could reflect a word, several words, or a concept.

Where are you seeing that? https://sbert.net/docs/package_reference/sentence_transformer/SentenceTransformer.html#sentence_transformers.SentenceTransformer.get_max_seq_length

Reply:
Okay, if they're comparable, then the code is equivalent.

Reply:
> Where are you seeing that?

Just using my prior NLP experience. I'll dig into their docs to find a relationship between `tokenize()` and `encode()`.
Consider this comment nonblocking.

Reply:
tokenizer.tokenize gives you the underling tokens:

```
>>> embedding_model.tokenizer.tokenize("Hello world.")
['hello', 'world', '.']
```

encode gives the the actual embedding:

```
>>> embedding_model.encode("Hello world.")
array([ 6.10581562e-02,  3.09772529e-02,  1.49336702e-03,  1.42844126e-03,
...
```

Reply:
Yea, to me, their lengths are not comparable, i.e, a float does not necessarily map one-to-one to a token.

Reply:
max_seq_length is the maximum number of tokens that can be used to generate the embedding. The size of the embedding is unrelated. E.g., for mpnet, the maximum number of tokens that can be provided as input to the embedding model is 512, while the size of the embeddings that mpnet produces is 768

Reply:
Ah, okay. My interpretation of `max_seq_length` is incorrect then. I thought it was characterizing the output embedding size.

Reply:
Totally misread this:
> Returns the maximal sequence length that the model _accepts_. Longer _inputs_ will be truncated.

Reply:
Yeah, max_seq_length is not the name I would have picked 😅